python/ray/data/dataset.py

```diff
@@ -5195,12 +5196,19 @@ def types(self) -> List[Union[type[object], "pyarrow.lib.DataType"]]:
         arrow_types = []
         for dtype in self.base_schema.types:
             if isinstance(dtype, TensorDtype):
+
+                if DataContext.get_current().use_arrow_tensor_v2:
```

Comment:
we should pass the data context from `Dataset.context` to the Schema constructor.
Otherwise, the following usage will break.
```
DataContext.set_current().use_arrow_tensor_v2 = True
ds1 = ... 
DataContext.set_current().use_arrow_tensor_v2 = False
ds2 = ... 
print(ds1.schema()) # should be True here
```

Reply:
Understood